src/Symfony/Component/Messenger/Transport/KernelTerminate/KernelTerminateTransportFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\KernelTerminate;
+
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class KernelTerminateTransportFactory
```

Comment:
`SymfonyKernelEventTransportFactory`?

Reply:
For the transport itself at least, I'd rather suggest [`MemoryTransport`](https://github.com/symfony/symfony/issues/28646#issuecomment-429112291) as nothing in the implementation is tied to a Symfony kernel event at all. One could use it in another way.
(I'd also remove the `KernelTerminate` sub-namespace of course).

But the factory can keep this name I guess.